src/platform/forms-system/src/js/components/ContactInfo.jsx

```diff
@@ -0,0 +1,397 @@
+      }
+    },
+  };
+
+  useEffect(
+    () => {
+      if (
+        (keys.email && (contactInfo.email?.emailAddress || '') !== email) ||
+        (keys.homePhone &&
+          contactInfo.homePhone?.updatedAt !== homePhone?.updatedAt) ||
+        (keys.mobilePhone &&
+          contactInfo.mobilePhone?.updatedAt !== mobilePhone?.updatedAt) ||
+        (keys.address &&
+          contactInfo.mailingAddress?.updatedAt !== address?.updatedAt)
+      ) {
+        const wrapper = { ...data[keys.wrapper] };
+        if (keys.email) {
+          wrapper[keys.email] = contactInfo.email?.emailAddress;
+        }
+        if (keys.address) {
+          wrapper[keys.address] = contactInfo.mailingAddress;
+        }
+        if (keys.homePhone) {
+          wrapper[keys.homePhone] = contactInfo.homePhone;
+        }
+        if (keys.mobilePhone) {
+          wrapper[keys.mobilePhone] = contactInfo.mobilePhone;
+        }
+        if (keys.email) {
+          wrapper[keys.email] = contactInfo.email?.emailAddress;
+        }
+        setFormData({ ...data, [keys.wrapper]: wrapper });
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [contactInfo, setFormData, data, keys],
+  );
+
+  useEffect(
+    () => {
+      if (editState) {
+        const [lastEdited, returnState] = editState.split(',');
+        setTimeout(() => {
+          const target =
+            returnState === 'canceled'
+              ? `#edit-${lastEdited}`
+              : `#updated-${lastEdited}`;
+          scrollTo('topScrollElement');
+          focusElement(target);
+        });
+      }
+    },
+    [editState],
+  );
+
+  useEffect(
+    () => {
+      if (missingInfo.length) {
+        // page had an error flag, so we know when to show a success alert
+        setHadError(true);
+      }
+    },
+    [missingInfo],
+  );
+
+  const MainHeader = onReviewPage ? 'h4' : 'h3';
+  const Headers = onReviewPage ? 'h5' : 'h4';
+  const headerClassNames = ['vads-u-font-size--h4', 'vads-u-width--auto'].join(
+    ' ',
+  );
+
+  const showSuccessAlert = (id, text) => {
+    // keep alerts in DOM, so we don't have to delay focus; but keep the 100ms
+    // delay to move focus away from the h3
+    const isHidden =
+      editState === `${id},updated` ? '' : 'vads-u-display--none';
+    return (
+      <va-alert
+        id={`updated-${id}`}
+        class={`vads-u-margin-y--1 ${isHidden}`}
+        status="success"
+        background-only
+        role="alert"
+      >
+        {`${text} ${content.updated}`}
+      </va-alert>
+    );
+  };
+
+  const homePhoneString = getPhoneString(dataWrap[keys.homePhone]);
+  const mobilePhoneString = getPhoneString(dataWrap[keys.mobilePhone]);
+  const editText = content.edit.toLowerCase();
+
+  // Loop to separate pages when editing
+  // Each Link includes an ID for focus managements on the review & submit page
+  const contactSection = [
+    keys.homePhone ? (
+      <React.Fragment key="home">
```

Comment:
For my own education, what is the use for using a key `home` on this fragment? Is it bc of the same reason you use keys within map loops?

Reply:
Yeah same reason - I'm adding JSX into an array, which will be rendered, so React likes to complain about a missing `key`